### PR TITLE
System/bank

### PR DESCRIPTION
**Pull request by sctnightcore**
```
bank
Enter command to use Bank System: bank <check || withdraw [zeny] || deposit
[zeny]>
bank check
-----------------------[Bank Storage]-----------------------
In Bank : (500) Zeny
On Hand : (55001000) Zeny
------------------------------------------------------------
bank withdraw 10
You gained 10 zeny.
[BankStatus:Withdraw]Balance in the bank: [490] Zeny
You lost 10 zeny.
[BankStatus:Deposit]Balance in the bank: [500] Zeny
-----------------------[Bank Storage]-----------------------
In Bank : (500) Zeny
On Hand : (55001000) Zeny
------------------------------------------------------------
```
Support:
* Bank check
* bank withdraw [Zeny]
* bank deposit [Zeny]
TODO:

- [x]  add type for support reason
- [ ]  Update Commands.pm to better in bank system

### CODE REVIEW

File: src/Network/Receive/kRO/Sakexe_0.pm [hunk 1]
```diff
@@ -379,9 +379,9 @@ sub new {
 		'098A' => ['clan_info', 'v a4 Z24 Z24 Z16 C2 a*', [qw(len clan_ID clan_name clan_master clan_map alliance_count antagonist_count ally_antagonist_names)]],
 		'098D' => ['clan_leave'],
 		'098E' => ['clan_chat', 'v Z24 Z*', [qw(len charname message)]],
-		'09A6' => ['banking_status', 'V2 C2',[qw(zeny reason)]],
-		'09A8' => ['banking_status_deposit', 'v V2 v2',[qw(reason inbank onhand)]],
-		'09AA' => ['banking_status_withdraw', 'v V2 v2',[qw(reason inbank onhand)]],
+		'09A6' => ['banking_status', 'V2 v',[qw(zeny reason)]],
+		'09A8' => ['banking_status_deposit', 'v V2 V',[qw(reason inbank onhard)]],
+		'09AA' => ['banking_status_withdraw', 'v V2 V',[qw(reason inbank onhard)]],
```
Comment (Nipodemos) [inline src/Network/Receive/kRO/Sakexe_0.pm]: onhard?
sorry what that means?

**Comment by moneyisgod** [comment]
In your log are missing the bank deposit command, since you lost 10 zenys ^^ !

**Comment by Nipodemos** [comment]
do you have a server on which i can test this ??

**Comment by sctnightcore** [comment]
@Nipodemos  Noo~~, I dont have it

**Comment by hubertyang88** [comment]
Tested in cRo which recently have bank function , it works!
-----------------------[Bank Storage]-----------------------
In Bank : (0) Zeny 
On Hand : (17556) Zeny 
------------------------------------------------------------
bank deposit 5
You lost 5 zeny.
[BankStatus:Deposit] : Success | [Bank Storage]: 5 Zeny.
bank withdraw 5 
You gained 5 zeny.
[BankStatus:Withdraw] : Success | [Bank Storage] : 0 Zeny.

**Comment by Nipodemos** [comment]
@sctnightcore what do you think? if you restore your branch i can merge it

**Comment by alisonrag** [comment]
@Nipodemos
This pull dont work for values above 2b (4bytes)

**Pull request closed without merging**